src/components/CreditCardBilling/CreditCardBilling.tsx

```diff
@@ -0,0 +1,299 @@
+    const width =
+      accumulativeValue * (datum.value / accumulativeValue) * reRangedXMaxRatio;
+    const start = i === 0 ? rectStart : rectStart + padding / 2;
+    const end = rectStart + width - paddingAdjustment;
+    const percentage = parseFloat(
+      ((datum.value / accumulativeValue) * 100).toFixed(1)
+    );
+    withCoordinates.push({
+      ...datum,
+      x: { start, end },
+      y: {
+        start: margin.top,
+        end: yMax - margin.top - margin.bottom,
+      },
+      percentage,
+    });
+    rectStart = end + paddingAdjustment;
+  });
+
+  return withCoordinates;
+};
+
+const drawChart = ({
+  canvasRef,
+  containerRef,
+  data,
+  margin = defaultMargins,
+  padding = 4,
+  colorPallete = pallete,
+}: {
+  canvasRef: HTMLCanvasElement;
+  containerRef: HTMLDivElement;
+  data: Record<'data', CreditCardBilling[]>;
+  margin?: Sides;
+  padding?: number;
+  colorPallete?: Record<string, React.CSSProperties['color']>;
+}) => {
+  const containerDimension = containerRef.getBoundingClientRect();
+  const withCoordinates = getCoordinatesAndPercentages({
+    data: data.data.sort((a, b) => {
+      return b.value - a.value;
+    }),
+    xMax: containerDimension.width,
+    yMax: containerDimension.height,
+    padding,
+    margin,
+  });
+
+  const ctx = canvasRef.getContext('2d');
+  if (ctx === null) return;
+
+  withCoordinates.forEach((datum, i) => {
+    const { x, y } = datum;
+
+    if (i === 0 || i === withCoordinates.length - 1) {
+      ctx.fillStyle = `${colorPallete[datum.category]}`;
+      ctx.roundRect(x.start, y.start, x.end - x.start, y.end, [
+        i === 0 ? roundRectRadii : 0,
+        i === withCoordinates.length - 1 ? roundRectRadii : 0,
+        i === withCoordinates.length - 1 ? roundRectRadii : 0,
+        i === 0 ? roundRectRadii : 0,
+      ]);
+      ctx.fill();
+    } else {
+      ctx.fillStyle = `${pallete[datum.category]}`;
+      ctx.fillRect(x.start, y.start, x.end - x.start, y.end);
+    }
+  });
+};
+
+const selectDatum = ({
+  e,
+  containerRef,
+  data,
+  padding,
+  margin = defaultMargins,
+}: {
+  e: MouseEvent;
+  containerRef: HTMLDivElement;
+  data: Record<'data', CreditCardBilling[]>;
+  padding: number;
+  margin: Sides;
+}) => {
+  const containerDimension = containerRef.getBoundingClientRect();
+  const withCoordinates = getCoordinatesAndPercentages({
+    data: data.data.sort((a, b) => {
+      return b.value - a.value;
+    }),
+    xMax: containerDimension.width,
+    yMax: containerDimension.height,
+    padding,
+    margin,
+  });
+  const selectTarget = withCoordinates.find((datum) => {
+    if (
+      e.clientX >= datum.x.start &&
+      e.clientX <= datum.x.end &&
+      e.clientY - containerDimension.top >= datum.y.start &&
+      e.clientY - containerDimension.top <= datum.y.start + datum.y.end
+    ) {
+      return true;
+    }
+
+    return false;
+  });
+
+  return selectTarget !== undefined ? selectTarget : null;
```

Comment:
```suggestion
  return selectTarget ?? null;
```
nullish한 값을 예외처리할 때는 [nullish coalescing operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing)를 사용하는 것을 추천드립니다!